frontend/util.lua

```diff
@@ -299,4 +299,26 @@ function util.getMenuText(item)
     return text
 end
 
+-- from http://notebook.kulchenko.com/programming/fixing-malformed-utf8-in-lua with modification
+function util.fixUtf8(str, replacement)
+    local pos = 1
+    local len = #str
+    while pos <= len do
+        if     pos == str:find("[%z\1-\127]", pos) then pos = pos + 1
+        elseif pos == str:find("[\194-\223][\128-\191]", pos) then pos = pos + 2
+        elseif pos == str:find(       "\224[\160-\191][\128-\191]", pos)
+            or pos == str:find("[\225-\236][\128-\191][\128-\191]", pos)
+            or pos == str:find(       "\237[\128-\159][\128-\191]", pos)
+            or pos == str:find("[\238-\239][\128-\191][\128-\191]", pos) then pos = pos + 3
+        elseif pos == str:find(       "\240[\144-\191][\128-\191][\128-\191]", pos)
+            or pos == str:find("[\241-\243][\128-\191][\128-\191][\128-\191]", pos)
+            or pos == str:find(       "\244[\128-\143][\128-\191][\128-\191]", pos) then pos = pos + 4
+        else
+            str = str:sub(1, pos - 1) .. replacement .. str:sub(pos + 1)
+            pos = pos + 1
```

Comment:
replacement can be more than one character, so you need to do `pos = pos + #replacement`